src/drivers/St7789.cpp

```diff
@@ -29,18 +29,32 @@ void St7789::Init() {
   DisplayOn();
 }
 
-void St7789::WriteCommand(uint8_t cmd) {
-  nrf_gpio_pin_clear(pinDataCommand);
-  WriteSpi(&cmd, 1);
+void St7789::WriteData(uint8_t data) {
+  WriteData(&data, 1);
 }
 
-void St7789::WriteData(uint8_t data) {
-  nrf_gpio_pin_set(pinDataCommand);
-  WriteSpi(&data, 1);
+void St7789::WriteData(const uint8_t* data, size_t size) {
+  WriteSpi(data, size, [pinDataCommand = pinDataCommand](bool isStart) {
+    if (isStart) {
+      nrf_gpio_pin_set(pinDataCommand);
+    }
+  });
+}
+
+void St7789::WriteCommand(uint8_t data) {
+  WriteCommand(&data, 1);
 }
 
-void St7789::WriteSpi(const uint8_t* data, size_t size) {
-  spi.Write(data, size);
+void St7789::WriteCommand(const uint8_t* data, size_t size) {
+  WriteSpi(data, size, [pinDataCommand = pinDataCommand](bool isStart) {
```

Comment:
To optimise more aggressively this hook could set the pin when `isStart` is false, and then the write data hook can be a no-op (nullptr)